kep/9-capacity-scheduling-for-ml-dl-workloads-based-on-scheduler-framework/README.md

```diff
@@ -0,0 +1,208 @@
+
+1. The resource buffer between "min" and "max" can help to tolerate runtime failures. For example, if a Pod that consumes the entire "min" fails to run, new Pods can still be created if the "max" has not been reached. When using Kubernetes resource quota, once the Pod that consumes the entire quota is created,
+no other Pods can be created - even if the Pod failed to be scheduled, or failed on running (error
+when pulling image, etc.).
+
+2. Improve overall resource utilization by allowing the workloads from one quota to "borrow" unused reserved "min" resources from other quotas. A quota's unused "min" resource can be used by other users, under the condition that there is a mechanism to guarantee the "victim" user can consume his "min" resource whenever he needs. Typically, this is done by implementing preemption.
+
+The success of “ElasticQuota” based capacity scheduling relies on a proper Pod preemption algorithm implementation. This KEP proposes the minimal scheduler extension to support the “ElasticQuota” based scheduling based on the scheduler framework.
+
+
+### Goals
+
+- The API definetion of ElasticQuota
+- Implement a scheduler plugin to honor the API of elastic resource quota.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+A "CRD" is needed to interact with the scheduler. The CRD kind name is subject to change. We use "ElasticQuota" in the proposal. The CRD defines the min and max allocation for the managed resources such as cpu, memory or extended resources. "ElasticQuota" is namespace scoped. A sample yaml is listed below:
+
+```yaml
+apiVersion: v1
+kind: ElasticQuota
+metadata:
+  name: test
+  namespace: test
+spec:
+  max:
+    cpu: 20
+    memory: 40Gi
+    nvidia.com/gpu: 2
+  min:
+    cpu: 10
+    memory: 20Gi
+    nvidia.com/gpu: 1
+```
+
+The definitions of "min" and "max" have been explained above. Max is infinite by default, min is 0 by default. The sum of min in all ElasticQuota must be less than or equal to the total resource capacity of the cluster.The ElasticQuota objects are created by a cluster administrator. One ElasticQuota object per namespace.
+
+### Relationship with ResourceQuota
+This version of proposal does not discuss merging the ElasticQuota into the existing ResourceQuota. But in the future, we will try to merge ElasticQuota into ResourceQuota when it's mature. We have three fields of resource in ElasticQuota(min max) and ResourceQuota(hard). 
```

Comment:
but min and max still need to respect the namespace's resourceQuota limit, i.e., `min <= max <= current resourceQuota`.  It is worth mentioning.